### ISSUE TITLE
Failure parse calendar with special symbol in TZID

### ISSUE THREAD

**Issue by slavag** [ical4j/ical4j#194]
Hi,
Sometime I'm getting calendars from office 365 with \u0015 or \u0017 as TZID value and 
parsing fails on DTSTART property.
Is this a bug in ical4j or such character is illegal by the spec ?

Thanks

**Comment by benfortuna** [comment]
Can you provide an example and I'll see if we can add support for such unicode characters.

**Comment by slavag** [comment]
Yes, please see the attached.
[cal.txt](https://github.com/ical4j/ical4j/files/1405929/cal.txt)

**Comment by slavag** [comment]
Hi, is there any progress on this ? As this issue is not rare at all in the Office 365 Calendars.
Thanks

**Comment by buckett** [comment]
I don't believe this is a valid TZID value.
Looking at the specification of the TZID value (https://tools.ietf.org/html/rfc5545#section-3.2.19) it can contain "paramtext" which is defined in content lines (https://tools.ietf.org/html/rfc5545#section-3.1)

    paramtext     = *SAFE-CHAR

    SAFE-CHAR     = WSP / %x21 / %x23-2B / %x2D-39 / %x3C-7E / NON-US-ASCII
    ; Any character except CONTROL, DQUOTE, ";", ":", ","

    WSP            =  SP / HTAB
    ; white space

    HTAB           =  %x09
    SP             =  %x20

    NON-US-ASCII  = UTF8-2 / UTF8-3 / UTF8-4
    ; UTF8-2, UTF8-3, and UTF8-4 are defined in [RFC3629]

Basically `UTF8-1 = %x00-7F` (lower ASCII characters) and is excluded from the NON-US-ASCII. Sets 2/3/4 are the higher sets.

`\u0015` is `%x15` and `\u0017` is `%x17` and these aren't valid paramtext characters and so technically they shouldn't be allowed in a TZID parameter.

**Comment by slavag** [comment]
I agree, but this is coming Office 365, so MS thinks it's valid :)

**Comment by benfortuna** [comment]
I think the best way to handle this would be to parse the invalid chars and convert to valid surrogates or just discard them. I did briefly try parsing without success, will have another go.
